sdk/spring-sdk/src/it/java/com/example/wiring/EventSourcedEntityIntegrationTest.java

```diff
@@ -0,0 +1,150 @@
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.Import;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static java.time.temporal.ChronoUnit.SECONDS;
+import static org.awaitility.Awaitility.await;
+import static org.junit.jupiter.api.Assertions.fail;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(classes = Main.class)
+@Import(KalixConfigurationTest.class)
+@TestPropertySource(properties = "spring.main.allow-bean-definition-overriding=true")
+public class EventSourcedEntityIntegrationTest {
+
+  @Autowired
+  private WebClient webClient;
+
+  private Duration timeout = Duration.of(10, SECONDS);
+
+  @Test
+  public void verifyCounterEventSourcedWiring() {
+
+    var counterId = "hello";
+
+    Integer counterIncrease = increaseCounter(counterId, 10);
+    Assertions.assertEquals(10, counterIncrease);
+
+    Integer counterMultiply = multiplyCounter(counterId, 20);
+    Assertions.assertEquals(200, counterMultiply);
+
+    int counterGet = getCounter(counterId);
+    Assertions.assertEquals(200, counterGet);
+  }
+
+  @Test
+  public void verifyCounterEventSourcedAfterRestart() {
+
+    var counterId = "helloRestart";
+
+    increaseCounter(counterId, 15);
+    multiplyCounter(counterId, 2);
+    int counterGet = getCounter(counterId);
+    Assertions.assertEquals(30, counterGet);
+
+    // force restart of counter entity
+    restartCounterEntity(counterId);
+
+    // events should be replayed successfully and
+    // counter value should be the same as previously
+    int counterGet2 = getCounter(counterId);
+    Assertions.assertEquals(30, counterGet2);
+  }
+
+  @Test
+  public void verifyCounterEventSourcedAfterRestartFromSnapshot() {
+
+    // snapshotting with kalix.event-sourced-entity.snapshot-every = 10
+    var counterId = "restartFromSnapshot";
+
+    // force the entity to snapshot - snapshots every 100
+    for (int i = 0; i < 10; i++) {
+      increaseCounter(counterId, 1);
+    }
+    Assertions.assertEquals(10, getCounter(counterId));
+
+    // force restart of counter entity
+    restartCounterEntity(counterId);
+
+    // current state is based on snapshot and should be the same as previously
+    Assertions.assertEquals(10, getCounter(counterId));
+  }
+
```

Comment:
testing the access to currentState after recovering from snapshot (which was actually failing as well) cc @aludwiko